### ISSUE TITLE
Should we add preprocessing layer in the model?

### ISSUE THREAD

**Issue by lynn901213** [tensorflow/cleverhans#1094]
Hi, 
Should we add preprocessing layer in the model for generating adversarial examples? If I want to attack pre-trained resnet50, the input preprocessing for training is caffe style and doesn't include normalization, so in order to get the right epsilon I always divide the input by 255.0 and add another two layers (x*255.0) and (resnet50.preprocessing_input) in the base model to ensure the input is in [0,1] range and preprocessing is also part of the attack. Is it the right the way to implement adversarial attacks?

**Comment by npapernot** [comment]
In the future, please ask questions that are not bug reports on stackoverflow. 

That's correct, you can either include the preprocessing layer in the model being attacked, or transform the adversarial examples obtained back to the original input layer by inverting the preprocessing transformation. 

See similar question: https://stackoverflow.com/questions/57097633/extracting-original-image-format-after-adversarial-attack-with-cleverhans

**Comment by lynn901213** [comment]
Thank you !!!!!!!!!!!!! And sorry I just posted another issue.
